extensions/vertx-http/dev-ui-resources/src/main/resources/dev-ui/qui/qui-d3-force.js

```diff
@@ -0,0 +1,285 @@
+import { select } from 'd3-selection';
+import { drag } from 'd3-drag';
+import { zoom, zoomIdentity } from 'd3-zoom';
+import ZoomEvent from 'd3-zoom/src/event.js';
+import 'qui-icon-menu';
+import '@vaadin/icon';
+
+/**
+ * Wrapping D3 Force 
+ */
+export class QuiD3Force extends LitElement {
+
+  static properties = {
+    title: { state: true },
+    nodes: { state: true },
+    links: { state: true },
+    types: { type: Array },
+  }
+
+  constructor() {
+    super();
+    this.title = "";
+    this.nodes = [];
+    this.links = [];
+    this.types = [];
+    this.draggedNode = null;
+    this.svg = null;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this.svg = select(this.shadowRoot.querySelector('svg'));
+
+    this.zoomBehavior = zoom()
+      .scaleExtent([0.1, 10])
+      .on('zoom', (e) => {
+        console.log("------> ZOOM !" + e);
+        const transform = new ZoomEvent().transform;
+        this.svg.select('#nodes').attr('transform', transform);
+        this.svg.select('#links').attr('transform', transform);
+      });
+
+    
+  }
+
+  updated() {
+    const { nodes, links, types } = this;
+
+    const width = this.clientWidth;
+    const height = this.clientHeight;
+    
+    const svg = select(this.shadowRoot.querySelector('svg'));
+
+    const linkGroup = svg.select('#links');
+    const nodeGroup = svg.select('#nodes');
+    const defsGroup = svg.select('#defs');
+
+    linkGroup.html("");
+    nodeGroup.html("");
+    defsGroup.html("");
+
+    defsGroup.selectAll("marker")
+    .data(types)
+    .join("marker")
+      .attr("id", d => `arrow-${d.name}`)
+      .attr("viewBox", "0 -5 10 10")
+      .attr("refX", 15)
+      .attr("refY", -0.5)
+      .attr("markerWidth", 8)
+      .attr("markerHeight", 8)
+      .attr("orient", "auto")
+      .append("path")
+      .attr("fill", d => d.color)
+      .attr("d", "M0,-5L10,0L0,5");
+
+    const link = linkGroup
+      .selectAll('path')
+      .data(links)
+      .join('path')
+      .attr('stroke', d => types.find(type => type.name === d.type).color)
+      .attr('fill', 'none')
+      .attr('marker-end', d => `url(#arrow-${d.type})`)
+      .attr('d', link => {
+        const dx = link.target.x - link.source.x;
+        const dy = link.target.y - link.source.y;
+        const dr = Math.sqrt(dx * dx + dy * dy);
+        return `M${link.source.x},${link.source.y}A${dr},${dr} 0 0,1 ${link.target.x},${link.target.y}`;
+      });
+
+    const node = nodeGroup
+      .selectAll('g')
+      .data(nodes, d => d.id)
+      .join('g')
+      .call(drag()
+        .on('start', event => {
+          this.draggedNode = event.subject;
+        })
+        .on('drag', event => {
+          this.draggedNode.x = event.x;
+          this.draggedNode.y = event.y;
+          this.simulation.alpha(0.5).restart();
+          this.updateNodePositions(node, link);
+        })
+        .on('end', () => {
+          this.draggedNode.fx = null;
```

Comment:
I think that we need to do something like above, i.e. `this.draggedNode.x = event.x;` and `this.draggedNode.y = event.y;`. Otherwise, the dragged node returns to its original position (that's at least what I observe locally...).